android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/CloudKeyboardMetaDataDownloadCallback.java

```diff
@@ -0,0 +1,349 @@
+  }
+
+  @Override
+  public void initializeContext(Context context) {
+
+  }
+
+  @Override
+  public List<CloudKeyboardMetaDataDownloadCallback.MetaDataResult> extractCloudResultFromDownloadSet(
+    CloudApiTypes.CloudDownloadSet<Void, List<CloudKeyboardMetaDataDownloadCallback.MetaDataResult>> aDownload) {
+
+    List<CloudKeyboardMetaDataDownloadCallback.MetaDataResult> _result = new ArrayList<>(aDownload.getSingleDownloads().size());
+
+    for (CloudApiTypes.SingleCloudDownload _d : aDownload.getSingleDownloads()) {
+
+      CloudApiTypes.CloudApiReturns _json_result = CloudDataJsonUtil.retrieveJsonFromDownload(_d);
+
+      if (_json_result!=null) {
+
+        CloudKeyboardMetaDataDownloadCallback.MetaDataResult _data = new CloudKeyboardMetaDataDownloadCallback.MetaDataResult();
+        _data.returnjson = _json_result;
+        _data.params = _d.getCloudParams();
+        _result.add(_data);  // Null if offline.
+      }
+    }
+    return _result;
+  }
+
+
+
+  @Override
+  public void applyCloudDownloadToModel(Context aContext, Void aModel, List<CloudKeyboardMetaDataDownloadCallback.MetaDataResult> aCloudResult)
+  {
+    if(aCloudResult.isEmpty()) {
+      String msg = aContext.getString(R.string.catalog_unavailable);
+      Toast.makeText(aContext, msg, Toast.LENGTH_SHORT).show();
+      //throw new IllegalStateException("Could not reach server");
+      return;
+    }
+
+    processCloudResults(aCloudResult);
+
+    startDownloads(aContext, aCloudResult);
+  }
+
+  /**
+   * Start the keyboard data and lexical model download.
+   * @param aContext the context
+   * @param aMetaDataResult the meta data result
+   */
+  private void startDownloads(Context aContext, List<MetaDataResult> aMetaDataResult) {
+    for(MetaDataResult _r:aMetaDataResult)
+    {
+      if(_r.additionalDownloads!=null)
+      {
+        if(_r.returnjson.target== CloudApiTypes.ApiTarget.Keyboard)
+        {
+          CloudKeyboardDataDownloadCallback _callback = new CloudKeyboardDataDownloadCallback();
+          _callback.setDownloadEventListeners(downloadEventListeners);
+          _callback.setKeyboardInfo(_r.keyboardInfo);
+
+          if(  CloudDownloadMgr.getInstance().alreadyDownloadingData(_r.additionalDownloadid))
+          {
+            continue;
+          }
+
+          KeyboardEventHandler.notifyListeners(downloadEventListeners,
+            KeyboardEventHandler.EventType.KEYBOARD_DOWNLOAD_STARTED, _r.keyboardInfo, 0);
+
+          CloudDownloadMgr.getInstance().executeAsDownload(aContext, _r.additionalDownloadid, null, _callback,
+            _r.additionalDownloads.toArray(new CloudApiTypes.CloudApiParam[0]));
+        }
+        else if(_r.returnjson.target== CloudApiTypes.ApiTarget.KeyBoardLexicalModels) {
+
+          if(  CloudDownloadMgr.getInstance().alreadyDownloadingData(_r.additionalDownloadid))
+          {
+            Toast.makeText(aContext,
+              aContext.getString(R.string.dictionary_download_is_running_in_background),
```

Comment:
When I install the **Salish, Straights --> SENCOTEN** keyboard I see this toast notification.

But I'm not seeing nrc.str.sencoten.model in the file system (appdata/models).
I would have expected the dictionary to also download and be enabled when I switch to the Sencoten keyboard.

Reply:
Fixed in https://github.com/keymanapp/keyman/pull/2275/commits/b9765970750fbe8461adcd7279298dfca01257a2